### DISCUSSION TITLE
Pipeling HTTP Requests

### DISCUSSION THREAD

**Discussion by hardillb** [fastify/fastify#5830] [General]
Hello,

We have an application that uses a HTTP authentication source (a POST to a known endpoint with username/password in JSON payload with some other attributes) and expects a JSON response with Accept/Reject for the user.

This app supports HTTP pipelining, e.g. sending multiple requests over the same HTTP connection, but not always waiting for a response before sending the next one.

Things get a lot better if we force the application to wait for a response before sending the next request.

All works well when client connect to the application in a staggered approach, but we have one specific group of users that like to restart all their clients at the same time causing a bit of a connection storm and this is bringing the Fastify app to it's knees.

We are investigating the backend database queries as well, but they all look pretty clean and optimised and the database doesn't appear to be struggling with the query load.

Is pipeling as described here something that should work well with Fastify?

**Comment by mcollina** [comment]
Yes, it should work as expected/codified in the rfc
